app/models/application_choice.rb

```diff
@@ -48,6 +48,18 @@ def offered_site
     offered_option.site
   end
 
+  def course_not_available?
+    course_option.course_not_available?
+  end
+
+  def course_full?
+    course_option.course_full?
+  end
+
+  def chosen_site_full?
+    course_option.no_vacancies?
+  end
```

Comment:
We try to keep our models slim... but I think that's more for "don't add 30 line methods to your models." These seem convenient. 👍